R/builder.R

```diff
@@ -94,10 +92,10 @@ add_layer <- function(.data,
     }
   }
 
-  out %>%
-    filter(!is.na(child),
-           !is.na(link),
-           !is.na(node_type))
+  layer <- out[!is.na(out$child) & !is.na(out$link) & !is.na(out$node_type), ]
```

Comment:
lol....im seeing my weird dependency on dplyr once you rewrite everything in base r.